src/palimpzest/query/processor/nosentinel_processor.py

```diff
@@ -215,6 +223,9 @@ def execute_plan(self, plan: PhysicalPlan, num_samples: int | float = float("inf
             if self.progress_manager:
                 self.progress_manager.finish()
 
+        logger.info(f"Done executing plan: {plan}")
+        logger.debug(f"Output records: {output_records}")
+        logger.debug(f"Plan stats: {plan_stats}")
```

Comment:
Same comment here as before re: selectively logging `plan_stats`; I also think you might have a similar issue with `output_records` if the list is very large.